File: backend_careercrafter/src/main/java/com/example/backend_careercrafter/controller/AuthController.java
package com.example.backend_careercrafter.controller;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.example.backend_careercrafter.dto.auth.AuthResponse;
import com.example.backend_careercrafter.dto.auth.LoginRequest;
import com.example.backend_careercrafter.dto.auth.RegisterRequest;
import com.example.backend_careercrafter.service.AuthService;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.http.MediaType;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;

@RestController
@RequiredArgsConstructor
@RequestMapping(path = "/api/v1/auth", produces = MediaType.APPLICATION_JSON_VALUE)
@Tag(name = "Authentication", description = "APIs for user registration and login")
public class AuthController {

    private final AuthService authService;

    @Operation(summary = "Register a new user")
    @PostMapping("/register")
    public ResponseEntity<AuthResponse> register(@Valid @RequestBody RegisterRequest request) {
        AuthResponse response = authService.register(request);
        return ResponseEntity.status(HttpStatus.CREATED).body(response);
    }

    @Operation(summary = "Login and get JWT token")
    @PostMapping("/login")
    public ResponseEntity<AuthResponse> login(@Valid @RequestBody LoginRequest request) {
        AuthResponse response = authService.login(request);
        return ResponseEntity.ok(response);
    }
}

File: python_console/auth/auth_controller.py
from auth.auth_service import AuthService


class AuthController:

    def __init__(self):
        self.auth_service = AuthService()

    def register(self, user):
        """Register a new user, handling duplicates and validation errors."""
        try:
            self.auth_service.register(user)
            print("\nRegistration successful! You can now login.")
        except Exception as e:
            print(f"Registration failed: {e}")

    def login(self, user):
        """Login an existing user and return a JWT token. Handles authentication errors."""
        try:
            token = self.auth_service.login(user)
            print("\nLogin Success")
            print("JWT Token:")
            print(token)
            return token
        except Exception as e:
            print(f"Login failed: {e}")
            return None